### PR TITLE
[go-gen] Add non-auth handler.go generation

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of handler.go for non-auth services

Once this is in, I think I've only got the main functions and the handlers left

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala [hunk 1]
```diff
@@ -1,19 +1,91 @@
...
-  )
+  private[service] def generateImports(root: ServiceRoot): String =
+    mkCode(
+      "import",
+      CodeWrap.parens.tabbed(
+        doubleQuote("errors"),
+        doubleQuote("fmt"),
+        doubleQuote("net/http"),
+        "",
+        s""""${root.module}/util"""",
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala]: Am I missing something, but can't this be `doubleQuote(s"${root.module}/util")`?

File: src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala [hunk 2]
```diff
@@ -1,19 +1,91 @@
...
+    )
+
+  private[service] def generateCommFunctionDecl(root: ServiceRoot, serviceName: String): String =
+    mkCode(
+      genFunctionCall(
+        s"Check${serviceName.capitalize}",
+        s"${serviceName}${root.idAttribute.name.toUpperCase} ${generateGoType(root.idAttribute.attributeType)}",
+        "token string",
+      ),
+      "(bool, error)",
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala:32]: I wonder if some argument labels would be useful here? Would make it easier to read. Haven't done it elsewhere tho so I guess would be a pain

**Review by jaylees14 — APPROVED** [review]
LGTM, just a couple of pedantic comments

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala]: whoops, that was the only import line I left in from the old code

File: src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/GoServiceCommGenerator.scala:32]: IntelliJ sticks them in, but not so readable on GitHub

**Pull request merged**